SplunkRumWorkspace/SplunkRum/SplunkRum/ScreenName.swift

```diff
@@ -52,6 +53,7 @@ func internal_setScreenName(_ newName: String, _ manual: Bool) {
     if oldName != nil && oldName! != "unknown" && (SplunkRum.configuredOptions?.screenNameSpans ?? true) {
         emitScreenNameChangedSpan(oldName!, newName)
     }
+    userDefaults.set(screenName, forKey: SCREEN_NAME_KEY)
```

Comment:
Should this perhaps be `newName` here? `screenName` is being accessed outside of a lock

Reply:
or at least protect the access

Reply:
We just save ScreenName in userdefaults nothing else. if we want to access screen name we have to use getScreenName().

Reply:
I added userDefaults.set(screenName, forKey: SCREEN_NAME_KEY) with in  Lock block.